client/host_queue.go

```diff
@@ -275,10 +275,13 @@ func (q *queue) asyncWrite(
 			q.Done()
 		}
 
+		hostID := q.host.ID() // NB(bl): this is passed to writeState to
+		// determine the state of the shard on the node we're writing to
```

Comment:
Might it be worth passing in the `topology.Host` so we have more context of the host that is completing a write request?